homework.cls

```diff
@@ -252,6 +268,9 @@
       \popQED
     \fi
     \endtrivlist\@endpefalse
+    \ifHW@pages
+      \clearpage
+    \fi
```

Comment:
This used to be implemented as a `newpage` option (cf. 4cca36dc4a7ecef753db867b04f6cf525e9c2e2e again). This won't make problems appear on a new page when only problem statements are included (e.g., if an instructor is using the class for posting problem sets). The old logic was maintaining a counter that increments with each problem statement, and clearing the page before starting a new problem after the first one.
https://github.com/artemmavrin/latex-homework/blob/68b423ff99e381a4bd091fab4dbb15f2960bbae5/homework.cls#L276-L280